camayoc/tests/qcs/test_scan.py

```diff
@@ -0,0 +1,185 @@
+        2) Send POST with data to create network profile using the host
+           credential to the profile endpoint.
+        3) Create a scan
+        4) Assert that the scan completes and a fact id is generated
+    :expectedresults: A scan is run and has facts associated with it
+    """
+    cfg = config.get_config()
+
+    auth_ids = []
+    for auth in cfg['qcs'].get('auths'):
+        if auth['name'] in profile['auths']:
+            cred = HostCredential(
+                client=shared_client,
+                ssh_keyfile=auth['sshkeyfile'],
+                username=auth['username'],
+            )
+            cred.create()
+            cleanup.append(cred)
+            auth_ids.append(cred._id)
+
+    netprof = NetworkProfile(
+        client=shared_client,
+        hosts=profile['hosts'],
+        credential_ids=auth_ids,
+    )
+    netprof.create()
+    scan = Scan(profile_id=netprof._id)
+    scan.create()
+    wait_until_complete(scan)
+    assert scan.status() == 'completed'
+    assert isinstance(scan.read().json().get('fact_collection_id'), int)
+    # add the ids to the lists to destroy after the test is done
+    cleanup.append(netprof)
+
+
+@pytest.mark.skip
+def test_pause(shared_client, cleanup, profile, isolated_filesystem):
+    """Run a scan on a system and confirm we can pause and cancel it.
+
+    :id: 4d4b9839-1672-4183-bd27-11864787eb8e
+    :description: Assert that scans can be paused and then cancelled.
+    :steps:
+        1) Create host credential
+        2) Send POST with data to create network profile using the host
+           credential to the profile endpoint.
+        3) Create a scan
+        4) Assert that the scan can be paused
+        5) Assert that the scan can be canceled
+    :expectedresults: A scan can be paused and cancelled
+    :caseautomation: notautomated
+    """
+    pass
+
+
+@pytest.mark.skip
+def test_restart(shared_client, cleanup, profile, isolated_filesystem):
+    """Run a scan on a system and confirm we can pause and restart it.
+
+    :id: 6d81121d-500e-4188-8195-2e469ca606c0
+    :description: Assert that scans can be paused and then restarted.
+    :steps:
+        1) Create host credential
+        2) Send POST with data to create network profile using the host
+           credential to the profile endpoint.
+        3) Create a scan
+        4) Assert that the scan can be paused
+        5) Assert that the scan can be restarted
+        6) Assert that he scan completes
+    :expectedresults: A restarted scan completes
+    :caseautomation: notautomated
+    """
+    pass
+
+
+@pytest.mark.skip
+def test_queue_mix_valid_invalid(
+        shared_client,
+        cleanup,
+        profile,
+        isolated_filesystem):
+    """Create a series of scans and assert all valid ones complete.
+
+    :id: eb6d07bd-39c5-4f9a-ae34-791c95f315fb
+    :description: Ensure scans can progress in queue to completion.
+    :steps:
+        1) Create a series of valid and invalid scans
```

Comment:
I am curious how to create invalid ones, do you mean having invalid data?

Reply:
Yes, creating a Scan object referring to non-existent hosts or credentials that don't exist or are missing information, then call create() on each.